kedro/framework/startup.py

```diff
@@ -19,13 +20,13 @@ class ProjectMetadata(NamedTuple):
     package_name: str
     project_name: str
     project_path: Path
-    project_version: str
+    kedro_init_version: str
```

Comment:
Since this is a factory method, `NamedTuple`, there doesn't seem to be a way to create a direct alias for its properties. Instead, I've kept both the `project_version` and `kedro_init_version` properties in `ProjectMetadata` as a temporary solution until 0.19. Both properties are treated the same way and given the same value with the only difference being that if `project_version` is found a deprecation warning is thrown. 

Do let me know your thoughts on this proposed solution.

Reply:
Could we then use `@dataclass(frozen=True)` instead then? Would that potentially break anything?

Reply:
Also `kedro_init_version` should go last, otherwise this is still a breaking change due to the order of arguments in the constructor.